src/Middleware/Debugger.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Debug\Api\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Router\UrlGeneratorInterface;
+use Yiisoft\Yii\Debug\DebuggerIdGenerator;
+
+final class Debugger implements MiddlewareInterface
+{
+    private DebuggerIdGenerator $idGenerator;
+    private UrlGeneratorInterface $urlGenerator;
+
+    public function __construct(DebuggerIdGenerator $idGenerator, UrlGeneratorInterface $urlGenerator)
+    {
+        $this->idGenerator = $idGenerator;
+        $this->urlGenerator = $urlGenerator;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $response = $handler->handle($request);
+        $duration = number_format((microtime(true) - $request->getAttribute('applicationStartTime')) * 1000 + 1);
```

Comment:
This is incorrect time without emit

Reply:
Final time should be collected on ApplicationShutdown event.

Reply:
WDYT do we need this header?

Reply:
Good question. Actually, I don't think we need it since duration info could be obtained from full data by ID, right?

Reply:
Yep. Let's remove it.